volatility3/framework/layers/objectstorage.py

```diff
@@ -0,0 +1,57 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import urllib.parse
+from typing import Optional, Any, List
+
+try:
+    import s3fs
+    HAS_S3FS = True
+except ImportError:
+    HAS_S3FS = False
+
+try:
+    import gcsfs
+    HAS_GCSFS = True
+except ImportError:
+    HAS_GCSFS = False
+
+from volatility3.framework import exceptions
+from volatility3.framework.layers import resources
+
+vollog = logging.getLogger(__file__)
+
+class S3FileSystemHandler(resources.VolatilityHandler):
+    if HAS_S3FS:
+        @classmethod
+        def non_cached_schemes(cls) -> List[str]:
+            return ["s3"]
+
+        @staticmethod
+        def default_open(req: urllib.request.Request) -> Optional[Any]:
+            """Handles the request if it's the s3 scheme."""
+            if req.type == "s3":
+                object_uri = "://".join(req.full_url.split("://")[1:])
```

Comment:
Hmmm, I know it's not actually a URL, but somehow feels like this should be handled by URL parsing code, in case people try to treat it like it is an actual URL?  I'd be happy if you just wanted to swap the scheme out with https or whatever the appropriate scheme is, but feels a little like it should have urllib involved here somehow, rather than a straight up text mangling, so that we get all the proper URL handling stuff for free?